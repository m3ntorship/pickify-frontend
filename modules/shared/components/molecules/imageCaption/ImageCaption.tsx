import React from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import type { IImageCaption } from './IImageCaption';
import styles from './ImageCaption.module.css';

const ImageCaption: FC<IImageCaption.IProps> = ({
  imgCaptionLetter,
  imgCaption,
}): ReactElement => {
  // make letter and devider (|) hidden when not pass letter as a props
  const captionLetterClasses = classNames({
    ' hidden ': !imgCaptionLetter,
  });
  return (
    <div className={styles.captionContainer}>
      <div className={styles.captionContent}>
        <span className={captionLetterClasses}>{imgCaptionLetter}</span>

        <div className={styles.captionDivider + captionLetterClasses} />

        <span>{imgCaption}</span>
      </div>
    </div>
  );
};

export default ImageCaption;
