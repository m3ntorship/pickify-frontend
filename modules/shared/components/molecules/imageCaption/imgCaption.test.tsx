import React from 'react';
import { render } from '@testing-library/react';
import ImageCaption from '.';

describe('imageCaption', () => {
  it('should render caption with text (Black T-shirt) and letter (A) when given this as property', () => {
    const { getByText } = render(
      <ImageCaption imgCaptionLetter="A" imgCaption="Black T-shirt" />,
    );
    expect(getByText('Black T-shirt')).toBeInTheDocument();
    expect(getByText('A')).toBeInTheDocument();
  });

  it('should render caption without letter and divider when not given this as a property', () => {
    const { getByTestId } = render(<ImageCaption imgCaption="Black T-shirt" />);
    expect(getByTestId('captionLetter')).toHaveClass('hidden');
    expect(getByTestId('captionDivider')).toHaveClass('hidden');
    // expect(getByTestId('captionLetter')).not.toBeVisible()
    // expect(getByTestId('captionDivider')).not.toBeVisible();
  });
});
