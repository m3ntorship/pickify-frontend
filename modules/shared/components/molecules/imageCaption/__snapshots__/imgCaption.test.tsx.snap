// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imageCaption snapshots should render caption with text (Black T-shirt) and letter (A) when given this as property 1`] = `
<div
  className="captionContainer"
>
  <div
    className="captionContent"
  >
    <span
      className=""
      data-testid="captionLetter"
    >
      A
    </span>
    <div
      className="captionDivider"
      data-testid="captionDivider"
    />
    <span>
      Black T-shirt
    </span>
  </div>
</div>
`;

exports[`imageCaption snapshots should render caption without letter and divider when not given this as a property 1`] = `
<div
  className="captionContainer"
>
  <div
    className="captionContent"
  >
    <span
      className=" hidden "
      data-testid="captionLetter"
    />
    <div
      className="captionDivider hidden "
      data-testid="captionDivider"
    />
    <span>
      Black T-shirt
    </span>
  </div>
</div>
`;
